bon/src/lib.rs

```diff
@@ -95,4 +96,52 @@ mod tests {
         let actual: Vec<String> = crate::vec![];
         assert!(actual.is_empty());
     }
+
+    #[test]
+    fn map_smoke() {
+        use std::collections::BTreeMap;
+        use std::collections::HashMap;
+
+        let hash_strings: HashMap<String, String> = crate::map! {
+            "Hello": "World",
+            "Goodbye": "Mars",
+        };
+
+        assert_eq!(hash_strings["Hello"], "World");
+        assert_eq!(hash_strings["Goodbye"], "Mars");
+
+        let tree_strings: BTreeMap<String, String> = crate::map! {
+            "Hello": "World",
+            "Goodbye": "Mars",
+        };
+
+        assert_eq!(tree_strings["Hello"], "World");
+        assert_eq!(tree_strings["Goodbye"], "Mars");
+    }
+
+    #[test]
+    fn set_smoke() {
+        use std::collections::BTreeSet;
+        use std::collections::HashSet;
+
+        let hash_strings: HashSet<String> = crate::set! {
+            "Hello", "World",
+            "Goodbye", "Mars",
+        };
```

Comment:
Let's use square brackets and propagate square brackets syntax for set in the docs as well. I believe `set![]` is closer to a `vec![]` in this regard.

```suggestion
        let hash_strings: HashSet<String> = crate::set![ 
            "Hello", "World",
            "Goodbye", "Mars",
        ];
```